public/services/DataSourceMenuContext.ts

```diff
@@ -0,0 +1,17 @@
+import { createContext } from "react";
+
+export interface DataSourceMenuProperties {
+  dataSourceId: string;
+  dataSourceLabel: string;
+  multiDataSourceEnabled: boolean;
+}
+
+const DataSourceMenuContext = createContext<DataSourceMenuProperties>({
+  dataSourceId: "",
+  dataSourceLabel: "",
```

Comment:
Still think the label is not necessary for the menuContext and in the url as it can be computed from list and id. What do you think?

Reply:
I agree that we can potentially remove it and compute from datasourceId. I will confirm after discussing with the opensearch-core team, as filters for the data sources are also being introduced this week, just to make sure functionality doesn't break.